llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -3507,16 +3507,39 @@ void SelectionDAGBuilder::visitInvoke(const InvokeInst &I) {
                           DAG.getBasicBlock(Return)));
 }
 
+/// The intrinsics currently supported by callbr are implicit control flow
+/// intrinsics such as amdgcn.kill.
+/// - they should be called (no "dontcall-" attributes)
+/// - they do not touch memory on the target (= !TLI.getTgtMemIntrinsic())
```

Comment:
Can we assert at least this one? To fail more obviously when people try to extend this to new intrinsics.